### DISCUSSION TITLE
Trabalhando com Wordpress em 2018 (Ou alternativa a altura?)

### DISCUSSION THREAD

**Discussion by raphaguasta** [frontendbr/forum#915] [Perguntas e Respostas]
Pessoal!

Na era do **ReactJs x VueJS x AngularJS, API REST e GraphQL, Serverless, SaaS**, e esse monte de coisa legal, surgiu uma demanda em que aparentemente **Wordpress** é a melhor solução.

Preciso de uma aplicação WEB de conteúdo dinâmico e gerenciável (CMS), renderizado pelo server, com possibilidade de testes A/B, integração com ferramentas de analytics, e uma possível inteligência de conteúdo específico para determinados usuários _(user A vê conteúdo A, user B vê conteúdo B)_, e claro, com um poder de SEO.

Não há a possibilidade de desenvolver todo esse ecossistema, por isso busquei ferramentas que possam ajudar. E a alternativa mais madura e viável me parece ser Wordpress!

Há alguns anos não trabalho com Wordpress, então gostaria de saber de vcs!

- Ainda usam Wordpress? Se não, qual a alternativa e como funciona?

- Como é o workflow front end? Quais ferramentas, processos, frameworks usam?

- Como lidar com o desenvolvimento de temas com os `<?php wp_tags_etc ?>` ? Qual o melhor jeito de trabalhar com isso?

- Service Workers, acessibilidade, SEO, imagens webp, e tecnologias WEB mais modernas.. Costumam usar plugins? Desenvolver plugins? Ou codar soluções pontuais?

Valeu! 😄

**Comment by luizbills** [comment]
> Ainda usam Wordpress? Se não, qual a alternativa e como funciona?

Eu ainda uso. Me dá dinheiro e tem bastante demanda, tá ótimo =D
Mas recentemente ouvi falar do [Bolt](https://bolt.cm/) que parece ser uma boa alternativa em termos de "ferramenta moderna". Para mim a maior vantagem do WP é a sua comunidade.

> Como é o workflow front end? Quais ferramentas, processos, frameworks usam?

Eu sou mais o cara do backend, trabalho muito é direto com PHP corrigindo bugs e implementando soluções. Mas para criar temas e cuidar do front-end do WP eu recomendo fortemente você usar o [Timber](https://www.upstatement.com/timber/). E claro que você também pode (deve) incorporar um gulp/webpack da vida para te ajudar na automação.

> Como lidar com o desenvolvimento de temas com os <?php wp_tags_etc ?> ? Qual o melhor jeito de trabalhar com isso?

Respondi acima. Só usar o timber e ser feliz.

> Service Workers, acessibilidade, SEO, imagens webp, e tecnologias WEB mais modernas.. Costumam usar plugins? Desenvolver plugins? Ou codar soluções pontuais?

Essas que você citou são o tipo de coisa que pode variar de desenvolvedor para desenvolvedor. Por exemplo, acessibilidade nem todo mundo faz igual ou se quer faz. Então sempre acabam sendo coisas de fazer na "mão", pois é muito dificil ter plugins que agrade a muitos ou por ser coisas praticamente 100% frontend (que não necessita de um plugin e sim apenas que os arquivos sejam carregados).

---

Bonus: esses dias desenvolvi um [framework para criar páginas de configuração](https://github.com/luizbills/better-wordpress-admin-api), caso você precise.

---

Também mantenho esse [quadro no Trello](https://trello.com/b/2i2ddQCp/wordpress-resources) com plugins que uso/usei e dicas que vejo por aí.

**Comment by raphaguasta** [comment]
Show @luizbills, obrigado!!

Esse Bolt parece legal, realmente uma pena que a comunidade ainda nâo chega nem perto =(.

Já esse Timber acho que é uma ferramenta que pode ser muito útil no meu caso, uma das minhas preocupações era confusão de código ao desenvolver o front com os <?php>, esse cara traz uma proposta pra separar isso. Show!!

**Comment by Sophmrs** [comment]
O [ghost](https://ghost.org) é uma opção bem legal que é bem popular, vide 24k stars no github https://github.com/tryghost/Ghost, mas é em javascript.

Também tá ficando popular CMSs decoupled/headless, aonde o CMS só cuida do content management em si, não da apresentação, que você faz como quiser. Inclusive dá pra usar o wordpress headless (https://github.com/postlight/headless-wp-starter por exemplo).

O [awesome cms](https://github.com/postlight/awesome-cms/blob/master/README.md) tem uma boa lista de outras opções.

**Comment by luizbills** [comment]
@raphaguasta se quiser testar coisas novas, acabei de ver esse headless-CMS [Strapi](https://strapi.io/demo). Parece bem daora.

**Comment by bernardodiasc** [comment]
@luizbills o Strapi aparentemente tem um grande potencial, uns 10 dias atrás tentei montar um protótipo de Gatsby+Strapi seguindo esse tutorial https://medium.com/@strapi/building-a-static-blog-using-gatsby-and-strapi-8b5acfc82ad8, mas o Strapi não funcionou nem com reza braba. Acho que ainda está bem imaturo pra fazer produtos, to no aguardo das correções dos bugs pra poder tentar novamente, pois baseado na proposta, parece ser muito bom.

Há uma plataforma de CMS que acho muito dahora tambem, https://www.datocms.com, funciona integrando outros sistemas baseados em arquivos estáticos, parece muito foda. O que me desanimou de usar é que é uma plataforma e não um sistema que vc pode pegar e instalar.

**Comment by luizbills** [comment]
@bernardodiasc pode cre, agora que eu vi que ainda está em *alpha*. Mas ficarei olho, parece ter muito potencial.

**Comment by raphaguasta** [comment]
@bernardodiasc @luizbills Valeu galera!

Pois é, tem algumas opções que parecem bem legais por aí! Mas no final das contas, o único que me dá segurança para implementar num grande projeto ainda é o Wordpress. x.x

**Comment by didiraja** [comment]
Não sou dev de Wordpress, apesar de pretender ser um dia. Anyway, ouvi falar recentemente do [Wordpressify](http://www.wordpressify.co)

**Comment by felipechalreo** [comment]
Estranho ninguém falar do https://roots.io/ ... Estou querendo aprender mais sobre ele.

**Comment by luizbills** [comment]
@felipechalreo eu já ouvi falar, conheço até gente que usa. Mas **eu** ainda não vi uma vantagem muito real. Só de ler algo que tem **bootstrap** na descrição eu já desanimo (deve ser opcional, claro) kkkk Mas eu não trabalho com desenvolvimento de tema e sim de plugins, então talvez nunca vou saber se realmente presta.

**Comment by joaodfmota** [comment]
@luizbills É opcional sim, com a nova versão do [sage](https://roots.io/sage/) no momento da instalação você escolhe se deseja utilizar o bootstrap entre outras opções. Recomendo "gastar" alguns minutos para testar todas as soluções deles.

**Comment by felipechalreo** [comment]
@joaodfmota Você usa roots no Windows?

**Comment by joaodfmota** [comment]
@felipechalreo Apenas o Bedrock e o Sage, precisa de ajuda?

**Comment by viniciusdocanto** [comment]
Usei o roots.io antes deles mudarem o nome do tema do WP para Sage, atualmente uso um tema customizado na agência que trabalho, que começou com a base do roots.
